sdk/nodejs/resource.ts

```diff
@@ -338,7 +338,9 @@ export abstract class Resource {
     }
 
     // getProvider fetches the provider for the given module member, if any.
-    public getProvider(moduleMember: string): ProviderResource | undefined {
+    //
+    // TODO the signature change from ProviderResource to ProvideResourceReference here may break compilation.
+    public getProvider(moduleMember: string): ProviderResourceReference | undefined {
```

Comment:
Breaking change potentially since getProvider is public. Or does it also have to be exported to be usable in user code?